### PR TITLE
Feature/solvency check

### PR DESCRIPTION
**Pull request by mr-rooftop**

### CODE REVIEW

File: bot/service/thorchain_network_service.py [hunk 1]
```diff
@@ -7,41 +7,21 @@
```
Comment (mbartn) [inline bot/service/thorchain_network_service.py:8]: You know you did a good job when you made a file 30 lines shorter

File: bot/helpers.py [hunk 2]
```diff
@@ -402,6 +402,127 @@ def did_churn_happen(validator, local_node_statuses, highest_churn_status_since)
...
+    asgard_actual = {}
+
+    asgard_expected = get_asgard_json()
+    pool_addresses = get_request_json_thorchain(url_path=':8080/v1/thorchain/pool_addresses')
+    for chain_data in pool_addresses['current']:
+        chain = chain_data['chain']
+        if chain == 'BNB':
+            if chain not in asgard_actual:
+                asgard_actual[chain] = {"json": {}}
+            asgard_actual[chain]['json'] = get_binance_balance(chain_data['address'])
```
Comment (mbartn) [inline bot/helpers.py:416]:
```suggestion
    asgard_actual = defaultdict(lambda: {"json": {}})

    asgard_expected = get_asgard_json()
    pool_addresses = get_request_json_thorchain(url_path=':8080/v1/thorchain/pool_addresses')
    for chain_data in pool_addresses['current']:
        chain = chain_data['chain']
        if chain == 'BNB':
            asgard_actual[chain]['json'] = get_binance_balance(chain_data['address'])
```

**Review by mbartn — APPROVED** [review]

File: bot/service/thorchain_network_service.py [hunk 3]
```diff
@@ -122,49 +59,84 @@ def is_binance_node_healthy(binance_node_ip) -> bool:
...
+def get_asgard_json() -> dict:
+    url = ':8080/asgard.json' if DEBUG else f":1317/thorchain/vaults/asgard"
+    return get_request_json_thorchain(url_path=url)
 
-    if constants_response.status_code != 200:
-        raise BadStatusException(constants_response)
 
-    return constants_response.json()['int_64_values']['BlocksPerYear']
+def get_yggdrasil_json() -> dict:
+    url = ":8080/yggdrasil.json" if DEBUG else ":1317/thorchain/vaults/yggdrasil"
```
[the remark this answers is not in the document]
Comment (mr-rooftop) [reply bot/service/thorchain_network_service.py:81]: It's a white lie

**Pull request merged**